drivers/adc/ad400x/ad400x.c

```diff
@@ -61,22 +61,17 @@ int32_t ad400x_spi_reg_read(struct ad400x_dev *dev,
 {
 	int32_t ret;
 	uint8_t buf[2];
-	uint32_t spi_clk_hz_tmp = (uint32_t) -1;
 
 	buf[0] = AD400X_READ_COMMAND;
 	buf[1] = 0xFF;
 
 	// register access runs at a lower clock rate (~2MHz)
-	if (dev->spi_desc->spi_clk_hz > dev->spi_desc->spi_clk_hz_reg_access) {
-		spi_clk_hz_tmp = dev->spi_desc->spi_clk_hz;
-		dev->spi_desc->spi_clk_hz = dev->spi_desc->spi_clk_hz_reg_access;
-	}
+	spi_engine_set_speed(dev->spi_desc, dev->reg_access_speed);
 
 	ret = spi_write_and_read(dev->spi_desc, buf, 2);
-	*reg_data = buf[1];
+	*reg_data = buf[0];
```

Comment:
Why did you change this to 0? 
Can't you change then 
```
spi_write_and_read(dev->spi_desc, buf, 2);
``` 
to
```
uint8_t buff = AD400X_READ_COMMAND;
spi_write_and_read(dev->spi_desc, &buff , 1); 
*reg_data = buff;
```
?

Reply:
2 represents the number of read/writes on SPI engine. Bytes are received starting with position 0 in new SPI engine, so this is correct.